src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/BarcodedFolder.groovy

```diff
@@ -0,0 +1,31 @@
+package life.qbic.datamodel.datasets.datastructure.folders.nanopore
+
+import life.qbic.datamodel.datasets.datastructure.folders.DataFolder
+import life.qbic.datamodel.identifiers.SampleCodeFunctions
+
+/**
+ * Specialisation version of a Folder object.
+ * 
+ * The folder name must contain the QBiC barcode.
+ *
+ * @author: Sven Fillinger
+ */
+class BarcodedFolder extends DataFolder {
+
+    protected BarcodedFolder() {}
+
+    protected BarcodedFolder(String name, String relativePath, List children) {
+        super(name, relativePath, children)
+        validateName()
+    }
+
+    static BarcodedFolder create(String name, String relativePath, List children) {
+        new BarcodedFolder(name, relativePath, children)
+    }
+
+    private void validateName() {
+        if (!(this.name =~ SampleCodeFunctions.QBIC_SAMPLE_ID_SCHEMA)) {
+            throw new IllegalArgumentException("Name must match the QBiC barcode schema!")
+        }
+    }
```

Comment:
```suggestion
    private static boolean isValidName(String name) {
        return (this.name =~ SampleCodeFunctions.QBIC_SAMPLE_ID_SCHEMA)
    }
```
I would throw the exception where the argument actually is illegal. The method validateName() does not take any argument.

Reply:
nope. tell don't ask